File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        
        int nonIndex =0;

        int n = nums.size();

        for(int i =0; i< n ; i++){

            if(nums[i] != 0){

                nums[nonIndex] = nums[i];
                nonIndex++;
            }
        }

        for(int i = nonIndex ; i < n ; i++){
            nums[i] = 0;
        }


    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
         
     List<Integer> tempList = new ArrayList<>();

     for(int i:nums){

        if(i != 0){
            tempList.add(i);
        }
     }

     for(int i =0; i<nums.length; i++){

         if(i < tempList.size()){
            nums[i] = tempList.get(i);
            continue;
         }

         nums[i] = 0;
     }
         
    }
}